### PR TITLE
Add readme for sql server

### PR DESCRIPTION
**Pull request by lyndaidaii**
- [x] I've read the guidelines for [contributing](CONTRIBUTING.md) and seen the [walkthrough](https://youtu.be/9OMxy1wal1s?t=1869)
- [x] I've posted a comment on an issue with a detailed description of how I am planning to contribute and got approval from a member of the team
- [x] The code builds and tests pass locally (also verified by our automated build checks)
- [ ] Commit messages follow this format:
```
        Summary of the changes
        - Add readme file for Microsoft.EntityFrameworkCore.SqlServer nuget package [Microsoft.EntityFrameworkCore.SqlServer](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.SqlServer/8.0.0-preview.1.23111.4)

        Fixes #bugnumber
```
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Code follows the same patterns and style as existing code in this repo

Address: https://github.com/dotnet/efcore/issues/30467, https://github.com/dotnet/efcore/issues/26556

### CODE REVIEW

File: src/EFCore.SqlServer/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+Microsoft.EntityFrameworkCore.SqlServer is Microsoft SQL Server EF Core Database Provider, which enables you to use Entity Framework Core with Microsoft SQL Server (including Azure SQL Database).
```
Comment (ajcvickers) [inline src/EFCore.SqlServer/README.md:1]:
```suggestion
Microsoft.EntityFrameworkCore.SqlServer is the database provider for Microsoft SQL Server and Azure SQL. This providers allows you to use Entity Framework Core with Microsoft SQL Server and Azure SQL databases.
```

File: src/EFCore.SqlServer/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## Getting started
+
+Explain how to use your package, provide clear and concise getting started instructions, including any necessary steps.
```
Comment (ajcvickers) [inline src/EFCore.SqlServer/README.md:5]: This seems like placeholder text.

File: src/EFCore.SqlServer/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+### Prerequisites
+
+- Supported database Engines: Microsoft SQL Server (2012 onwards)
+- The provider references Microsoft.Data.SqlClient (not System.Data.SqlClient). If your project takes a direct dependency on SqlClient, make sure it references the Microsoft.Data.SqlClient package.
```
Comment (ajcvickers) [inline src/EFCore.SqlServer/README.md:10]: Make these links?

File: src/EFCore.SqlServer/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+Once you've installed the package, you can use it in your Entity Framework Core application by specifying the SQL Server provider in your DbContext's OnConfiguring method:
+
+```csharp
+protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+{
+    optionsBuilder.UseSqlServer("Server=(localdb)\\mssqllocaldb;Database=MyDatabase;Trusted_Connection=True;");
+}
+```
+
+In this example, we're using the (localdb)\mssqllocaldb server with the MyDatabase database. You'll need to adjust the connection string to match your own SQL Server instance and database.
```
Comment (ajcvickers) [inline src/EFCore.SqlServer/README.md:23]: Link to guidance on this?

File: src/EFCore.SqlServer/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+The SQL Server provider for Entity Framework Core supports a variety of features, including:
+
+- Basic CRUD operations
+- Transactions
+- Stored procedures
+- Views
+- Table-valued functions
+- Identity columns
+- In-memory tables
```
Comment (ajcvickers) [inline src/EFCore.SqlServer/README.md:35]:
```suggestion
The SQL Server provider  supports all common features of [Entity Framework Core](https://learn.microsoft.com/ef/core/) as well as some [SQL Server-specific features](https://learn.microsoft.com/ef/core/providers/sql-server/?tabs=dotnet-core-cli) including temporal tables and memory-optimized tables.
```

**Review by ajcvickers — CHANGES_REQUESTED** [review]
Looking food; needs a few updates.

**Review by ajcvickers — APPROVED** [review]

**Comment by ErikEJ** [comment]
Why is there a section about .Design??

**Comment by ErikEJ** [comment]
I was looking at the wrong branch, my bad.

**Pull request merged**